DotNetCoreMasters/DotNetCoreMasters/Controllers/ItemsController.cs

```diff
@@ -8,16 +12,64 @@
 
 namespace DotNetCoreMasters.Controllers
 {
-    [ApiController]
     [Route("[controller]")]
-    public class ItemsController : Controller
+    public class ItemsController : ControllerBase
     {
-        public IActionResult GetAll(int userId)
+        private readonly IItemService _itemService;
+
+        public ItemsController(IItemService itemService)
+        {
+            _itemService = itemService;
+        }
+
+        [HttpGet]
+        public IActionResult GetAll()
+        {
+            var listItems = _itemService.GetAll().ToList();
+
+            return Ok(listItems);
+        }
+
+        [HttpGet]
+        public IActionResult Get(int itemId)
+        {
+
+            return Ok(itemId);
+        }
+
+
+        [HttpGet]
+        public IActionResult GetByFilters([FromQuery] Dictionary<string, string> filters)
         {
-            var res = new ItemService();
-            var listOfStrings = res.GetAll(userId);
+            var filter = filters;
 
-            return Ok(listOfStrings);
+            return Ok(filter);
         }
+
+        [HttpPost]
+        public IActionResult Post([FromBody] ItemCreateBindingModel itemCreateModel)
+        {
+            var model = itemCreateModel;
+
+            return Ok();
+        }
+
+        [HttpPut]
+        public IActionResult Put(int itemId, [FromBody] ItemCreateBindingModel itemCreateModel)
+        {
+            var newId = itemId;
+            var model = itemCreateModel;
+
+            return Ok();
+        }
+
+        [HttpDelete]
```

Comment:
route parameter? action name should be Delete

Reply:
still no route parameter on the Route attribute. make sure the itemId is same with other routes